app/code/Magento/ImportExport/Model/Import/SampleFileProvider.php

```diff
@@ -0,0 +1,128 @@
+     * @param ComponentRegistrar $componentRegistrar
+     * @param array $samples
+     */
+    public function __construct(
+        \Magento\Framework\Filesystem\Directory\ReadFactory $readFactory,
+        \Magento\Framework\Component\ComponentRegistrar $componentRegistrar,
+        array $samples = []
+    ) {
+        $this->readFactory = $readFactory;
+        $this->componentRegistrar = $componentRegistrar;
+        $this->samples = $samples;
+    }
+
+    /**
+     * Returns the Size for the given file associated to an Import entity
+     *
+     * @param string $entityName
+     * @throws NoSuchEntityException
+     * @return int
+     */
+    public function getSize(string $entityName): int
+    {
+        $directoryRead = $this->getDirectoryRead($entityName);
+        $filePath = $this->getPath($entityName);
+        $fileSize = isset($directoryRead->stat($filePath)['size'])
+            ? $directoryRead->stat($filePath)['size'] : null;
+
+        return $fileSize;
+    }
+
+    /**
+     * Returns Content for the given file associated to an Import entity
+     *
+     * @param string $entityName
+     * @throws NoSuchEntityException
+     * @return string
+     */
+    public function getFileContents(string $entityName): string
+    {
+        $directoryRead = $this->getDirectoryRead($entityName);
+        $filePath = $this->getPath($entityName);
+
+        return $directoryRead->readFile($filePath);
+    }
+
+    /**
+     * @param string $entityName
+     * @return string $entityName
+     * @throws NoSuchEntityException
+     */
+    private function getPath(string $entityName): string
+    {
+        $directoryRead = $this->getDirectoryRead($entityName);
+        $moduleName = $this->getModuleName($entityName);
+        $moduleDir = $this->componentRegistrar->getPath(ComponentRegistrar::MODULE, $moduleName);
+        $fileAbsolutePath = $moduleDir . '/Files/Sample/' . $entityName . '.csv';
+
+        $filePath = $directoryRead->getRelativePath($fileAbsolutePath);
+
+        if (!$directoryRead->isFile($filePath)) {
+            throw new NoSuchEntityException(__("There is no file: %file", ['file' => $filePath]));
+        }
+
+        return $filePath;
+    }
+
+    /**
+     * @param string $entityName
+     * @return ReadInterface
+     */
+    private function getDirectoryRead(string $entityName): ReadInterface
+    {
+        $moduleName = $this->getModuleName($entityName);
+        $moduleDir = $this->componentRegistrar->getPath(ComponentRegistrar::MODULE, $moduleName);
+        $directoryRead = $this->readFactory->create($moduleDir);
+
+        return $directoryRead;
+    }
+
+    /**
+     * @param string $entityName
+     * @return string
+     * @throws NoSuchEntityException
+     */
+    private function getModuleName(string $entityName): string
+    {
+        if (!isset($this->samples[$entityName])) {
+            throw new NoSuchEntityException();
```

Comment:
Why do we use `NoSuchEntityException` for missed array key?
Maybe do we need to use something like `WrongInputParameters`?